test/lib/migrationManager.test.js

```diff
@@ -0,0 +1,364 @@
+        var error = new Error('Some unexpected error')
+        error.code = 6000
+        error.status = 400
+        throw error
+      })
+
+      try {
+        await migrateAccount({
+          email: 'account@buffer.com',
+          password: 'password',
+        })
+      } catch (err) {
+        expect(mockAuthManager.getAccount).toBeCalledWith({ email: 'account@buffer.com' })
+        expect(mockReplayApi.verifyEmailAndPassword).toBeCalledWith({
+          email: 'account@buffer.com',
+          password: 'password',
+        })
+
+        expect(err.message).toBe('Some unexpected error')
+        expect(err.code).toBe(6000)
+        expect(err.status).toBe(400)
+      }
+    })
+
+    it('should throw an error while creating the account', async () => {
+      mockAuthManager.getAccount.mockImplementation(() => {
+        var error = new Error('Account can not be found')
+        error.code = 6003
+        error.status = 400
+        throw error
+      })
+
+      mockReplayApi.verifyEmail.mockReturnValue({
+        body: {
+          data: {
+            userId: 'replyFK'
+          }
+        }
+      })
+
+      mockCrypto.randomBytes.mockImplementation(() => {
+        return 'random-password'
+      })
+
+      mockReplayApi.verifyEmailAndPassword.mockReturnValue({
+        body: {
+          data: {
+            userId: 'replyFK'
+          }
+        }
+      })
+
+      mockAuthManager.createAccount.mockImplementation(() => {
+        var error = new Error('Some unexpected error')
+        error.code = 6000
+        error.status = 400
+        throw error
+      })
+      try {
+        await migrateAccount({
+          email: 'account@buffer.com'
+        })
+      } catch (err) {
+        expect(mockAuthManager.getAccount).toBeCalledWith({
+          email: 'account@buffer.com',
+        })
+        expect(mockReplayApi.verifyEmail).toBeCalled()
+        expect(mockCrypto.randomBytes).toBeCalled()
+        expect(mockAuthManager.createAccount).toBeCalled()
+
+        expect(err.message).toBe('Some unexpected error')
+        expect(err.code).toBe(6000)
+        expect(err.status).toBe(400)
+      }
+
+      try {
+        await migrateAccount({
+          email: 'account@buffer.com',
+          password: 'password',
+        })
+      } catch (err) {
+        expect(mockAuthManager.getAccount).toBeCalledWith({
+          email: 'account@buffer.com',
+        })
+        expect(mockReplayApi.verifyEmailAndPassword).toBeCalledWith({
+          email: 'account@buffer.com',
+          password: 'password',
+        })
+        expect(mockAuthManager.createAccount).toBeCalledWith({
+          email: 'account@buffer.com',
+          password: 'password',
+        })
+
+        expect(err.message).toBe('Some unexpected error')
+        expect(err.code).toBe(6000)
+        expect(err.status).toBe(400)
+      }
+    })
+
+    it('should throw an error while creating the product link', async () => {
```

Comment:
How about making the behavior more explicit in the test name:

```suggestion
    it('should remove the created account if createProductLink fails', async () => {
```